indexing-service/src/main/java/org/apache/druid/indexing/seekablestream/supervisor/SeekableStreamSupervisorSpec.java

```diff
@@ -151,6 +157,37 @@ public DruidMonitorSchedulerConfig getMonitorSchedulerConfig()
   @Override
   public abstract Supervisor createSupervisor();
 
+  /**
+   * need to notice that autoScaler would be null which means autoscale is dissable.
+   * @param supervisor
+   * @return autoScaler, disable autoscale will return dummyAutoScaler and enable autoscale wiil return defaultAutoScaler by default.
+   */
+  @Override
+  @SuppressFBWarnings(value = "RV_RETURN_VALUE_IGNORED", justification = "using siwtch(String)")
+  public SupervisorTaskAutoscaler createAutoscaler(Supervisor supervisor)
```

Comment:
The way `autoScaler` instance is created here does not support custom autoScaler implementation in extensions as switch statement is used to create the instance. If a new strategy is implemented for autoScaler in an extension, this class needs to be changed to support it which is not ideal as its a change in core Druid. I have raised a PR on your branch on how we can fix this - https://github.com/zhangyue19921010/druid/pull/1

The changes are - 
1. Add `SupervisorTaskAutoScaler createAutoScaler(Supervisor supervisor, SupervisorSpec spec);` method in `AutoScalerConfig` that will be called from `SeekableStreamSupervisorSpec` to create autoScaler.
2. I don't think `getAutoScalerStrategy` method is needed in `AutoScalerConfig` as implementation of `AutoScalerConfig` can return instance of `AutoScaler` directly on call to `createAutoScaler`.

Reply:
Nice Coding! changed.